Makefile

```diff
@@ -301,6 +301,7 @@ VIRTENV='virtualenv==20.21.0'
 # https://stackoverflow.com/questions/63515454/why-does-pip3-install-pipenv-give-error-error-cannot-uninstall-distlib
 install-deps-ALPINE-for-pysemgrep:
 	apk add --no-cache $(ALPINE_APK_DEPS_PYSEMGREP)
+	pip uninstall -y virtualenv
```

Comment:
I will leave a comment, I didn't get to elaborate on this because I abruptly had to stop working due to my flight.

TL;DR there is a host of nasty, messy `virtualenv` errors we get from the following line (which simply installs `virtualenv`). The linked Stackoverflow post seems to indicate this is due to a conflict between a previously-installed version of `virtualenv`.

I investigated the Docker images and both the working image and the newer image do have `virtualenv` installed (which originates as a dependency when installing `pre-commit`, but the disparity is in their versioning. I tried a great deal of options (including force installing, installing `virtualenv-clone`, not installing the deps, etc), but the only option that seemed to work was uninstalling `virtualenv` entirely prior to reinstalling it.

I viewed this as the lowest friction solution. I plan to document this better and proceed like this, but if you have a better suggestion, I would love to do something different.

Reply:
Gotcha, I agree it's not worth spending the calories right now. A brief clarifying comment sounds good to me 👍